torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -889,27 +889,24 @@ def arg_extractor(input, dim, combine_fn):
 
         input, dim, combine_fn = arg_extractor(*args, **kwargs)
 
-        assert isinstance(
```

Comment:
why?

Reply:
The use of `functools.partial` to wrap the `combine_fn` resulted in this becoming `FunctoolsPartialVariable`.